app_dart/lib/src/service/cache_service.dart

```diff
@@ -56,42 +56,40 @@ class CacheService {
     Future<Uint8List> Function()? createFn,
     Duration ttl = const Duration(minutes: 1),
   }) async {
-    
-      final Cache<Uint8List> subcache = cache.withPrefix(subcacheName);
-      Uint8List? value;
-
-      await m.acquireRead();
-      try {
-        print("Acquiring getOrCreate read lock.");
-        value = await subcache[key].get();
-      } catch (e) {
-        if (attempt < maxCacheGetAttempts) {
-          return getOrCreate(
-            subcacheName,
-            key,
-            attempt: ++attempt,
-            createFn: createFn,
-            ttl: ttl,
-          );
-        } else {
-          // Give up on trying to get the value from the cache.
-          value = null;
-        }
-      }  finally {
-        print("Releasing getOrCreate read lock.");
-        m.release();
-      }
+    final Cache<Uint8List> subcache = cache.withPrefix(subcacheName);
+    Uint8List? value;
 
-      // If given createFn, update the cache value if the value returned was null.
-      if (createFn != null && value == null) {
-        // Try creating the value
-        value = await createFn();
-        print("Acquiring getOrCreate write lock.");
-        await set(subcacheName, key, value, ttl: ttl);
+    await m.acquireRead();
+    try {
+      print("Acquiring getOrCreate read lock.");
+      value = await subcache[key].get();
+    } catch (e) {
+      if (attempt < maxCacheGetAttempts) {
+        return getOrCreate(
+          subcacheName,
+          key,
+          attempt: ++attempt,
+          createFn: createFn,
+          ttl: ttl,
+        );
+      } else {
+        // Give up on trying to get the value from the cache.
+        value = null;
       }
+    } finally {
+      print("Releasing getOrCreate read lock.");
```

Comment:
s/print/logging?

Reply:
You can disregard these. They will get removed.